lib/jxl/color_encoding_internal.cc

```diff
@@ -106,8 +106,8 @@ std::string ToString(RenderingIntent rendering_intent) {
             static_cast<uint32_t>(rendering_intent));
 }
 
-static double F64FromCustomxyI32(const int32_t i) { return i * 1E-6; }
-static Status F64ToCustomxyI32(const double f, int32_t* JXL_RESTRICT i) {
+static double F64FromCustomxyI32(int32_t i) { return i * 1E-6; }
```

Comment:
Why does VS complain about the const here? it *does* have an effect, so I'd rather disable the VS warning...

Reply:
(also in many other places)

Reply:
Ah, I didn't check which ones it complained about and which ones it didn't.

In terms of code behavior, a const in function arguments that are passed by value doesn't mean anything, does it?

Reply:
It doesn't mean anything on function *declarations*. On function definitions it's the same as declaring a variable const.

Reply:
Yes, it could theoretically catch bugs if the intention is that the arg remains constant throughout the function body, but for simple one-statement functions like this one I don't really see the added value. For less trivial function bodies I agree it might be worthwhile to keep the const.

Reply:
Even so, I wouldn't remove it unless there's a good reason, and I don't think there's a good reason ;)

Reply:
Fair enough. Brought back the `const`s (and maybe added a few new ones in the process).

I think what MSVC was complaining about was just the `const`s in data_parallel.h, I probably misinterpreted the warning messages. See https://github.com/libjxl/libjxl/pull/1005